service/src/main/kotlin/no/nav/su/se/bakover/service/revurdering/RevurderingServiceImpl.kt

```diff
@@ -284,53 +284,86 @@ internal class RevurderingServiceImpl(
                             KunneIkkeBeregneOgSimulereRevurdering.SimuleringFeilet
                         }.map {
-                            val simulert = beregnetRevurdering.toSimulert(it.simulering)
+                            val simulert = opphør.toSimulert(it.simulering)
                             revurderingRepo.lagre(simulert)
                             simulert
                         }
                     }
+                    Resultat.Innvilget -> {
+                        when (
+                            val beregnetRevurdering = originalRevurdering.beregn(fradrag)
+                                .getOrHandle {
+                                    return when (it) {
+                                        is Revurdering.KunneIkkeBeregneRevurdering.KanIkkeVelgeSisteMånedVedNedgangIStønaden -> KunneIkkeBeregneOgSimulereRevurdering.KanIkkeVelgeSisteMånedVedNedgangIStønaden
+                                        is Revurdering.KunneIkkeBeregneRevurdering.UfullstendigBehandlingsinformasjon -> KunneIkkeBeregneOgSimulereRevurdering.UfullstendigBehandlingsinformasjon
+                                    }.left()
+                                }
+                        ) {
+                            is BeregnetRevurdering.IngenEndring -> {
+                                revurderingRepo.lagre(beregnetRevurdering)
+                                beregnetRevurdering.right()
+                            }
+                            is BeregnetRevurdering.Innvilget -> {
+                                utbetalingService.simulerUtbetaling(
+                                    sakId = beregnetRevurdering.sakId,
+                                    saksbehandler = saksbehandler,
+                                    beregning = beregnetRevurdering.beregning,
+                                ).mapLeft {
+                                    KunneIkkeBeregneOgSimulereRevurdering.SimuleringFeilet
+                                }.map {
+                                    val simulert = beregnetRevurdering.toSimulert(it.simulering)
+                                    revurderingRepo.lagre(simulert)
+                                    simulert
+                                }
+                            }
 
-                    is BeregnetRevurdering.Opphørt -> {
-                        utbetalingService.simulerOpphør(
-                            sakId = beregnetRevurdering.sakId,
-                            saksbehandler = saksbehandler,
-                            opphørsdato = beregnetRevurdering.periode.fraOgMed,
-                        ).mapLeft {
-                            KunneIkkeBeregneOgSimulereRevurdering.SimuleringFeilet
-                        }.map {
-                            val simulert = beregnetRevurdering.toSimulert(it.simulering)
-                            revurderingRepo.lagre(simulert)
-                            simulert
+                            is BeregnetRevurdering.Opphørt -> {
```

Comment:
Jeg har ikke insett konsekvensene av dette steget før tror jeg. For her vil vi både sende en ny beregning til oppdrag, samtidig som vi opphører. Mens dersom det er et opphør pgr. vilkårene, vil vi bruke nåværende oppdragslinjer innenfor revurderingsperioden, som (antageligvis) vil være lik som. Det vil vi kanskje kunne verifisere i simuleringa?

Reply:
Det kommer vel litt an på hvordan "nivå" vi snakker om/tenker på. Slik jeg ser det _ønsker_ vi i akkurat dette tilfellet å kun opphøre eksisterende linjer (det er slik funksjonaliteten for opphør fungerer p.t). Dersom du tenker på kombinasjoner av opphør og endret beregning i samme revurdering er jeg enig i at det må gjøres noe.